libnavigation-core/src/test/java/com/mapbox/navigation/core/RoutesProgressDataProviderTest.kt

```diff
@@ -0,0 +1,143 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.routealternatives.AlternativeMetadataProvider
+import com.mapbox.navigation.core.routealternatives.AlternativeRouteMetadata
+import com.mapbox.navigation.core.routealternatives.AlternativeRouteProgressDataProvider
+import com.mapbox.navigation.testing.MainCoroutineRule
+import io.mockk.coEvery
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkObject
+import io.mockk.unmockkObject
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class RoutesProgressDataProviderTest {
+
+    @get:Rule
+    val mainCoroutineRule = MainCoroutineRule()
+
+    private val primaryRouteProgressDataProvider =
+        mockk<PrimaryRouteProgressDataProvider>(relaxed = true)
+    private val alternativeMetadataProvider = mockk<AlternativeMetadataProvider>(relaxed = true)
+    private val sut = RoutesProgressDataProvider(
+        primaryRouteProgressDataProvider,
+        alternativeMetadataProvider
+    )
+
+    @Before
+    fun setUp() {
+        mockkObject(AlternativeRouteProgressDataProvider)
+    }
+
+    @After
+    fun tearDown() {
+        unmockkObject(AlternativeRouteProgressDataProvider)
+    }
+
+    @Test(expected = IllegalArgumentException::class)
+    fun `getRoutesProgressData for empty routes`() = mainCoroutineRule.runBlockingTest {
+        sut.getRoutesProgressData(emptyList())
+    }
```

Comment:
Just a matter of taste. Why do you need to set test main dispatcher here? Maybe it's simpler without it? 
```suggestion

    private val primaryRouteProgressDataProvider =
        mockk<PrimaryRouteProgressDataProvider>(relaxed = true)
    private val alternativeMetadataProvider = mockk<AlternativeMetadataProvider>(relaxed = true)
    private val sut = RoutesProgressDataProvider(
        primaryRouteProgressDataProvider,
        alternativeMetadataProvider
    )

    @Before
    fun setUp() {
        mockkObject(AlternativeRouteProgressDataProvider)
    }

    @After
    fun tearDown() {
        unmockkObject(AlternativeRouteProgressDataProvider)
    }

    @Test(expected = IllegalArgumentException::class)
    fun `getRoutesProgressData for empty routes`() = runBlockingTest {
        sut.getRoutesProgressData(emptyList())
    }
```